mods/persistence/modules/recycling/recycler.dm

```diff
@@ -0,0 +1,383 @@
+		/decl/public_access/public_variable/reagents/total_volume,
+		/decl/public_access/public_variable/reagents/maximum_volume,
+	)
+	public_methods = list(
+		/decl/public_access/public_method/toggle_power,
+	)
+
+	var/bloodied 				= FALSE		//Whether we show the sprite with blood or not
+	var/tmp/grinding			= FALSE 	//Whether we're currently busy grinding something down
+	var/tmp/safety_mode 		= 0 		// Temporality stops the machine if it detects a mob
+	var/tmp/efficiency 			= 0.5		//Percentage of materials recovered from the recycled item
+	var/tmp/timelastoutput 		= 0			//Time when the last sheet was outputed from the machine
+	var/const/sound_idle 		= 'sound/machines/creaky_loop.ogg'
+	var/const/sound_processing 	= 'sound/machines/machine_wirr.ogg'
+	var/const/sound_outputs 	= 'sound/machines/hiss.ogg'
+	var/const/sound_reject 		= 'sound/machines/buzz-sigh.ogg'
+	var/tmp/soundid 			= "Recyclers"
+	var/tmp/datum/sound_token/sound_looping
+	var/eat_dir = WEST						//Fixed directio from where the machine will accept trash
+	var/list/stored_material = list()		//List of /decl/materials paths and matter unit value to keep track of all the matter that was processed. (upgrade when material states are less fucky)
+
+
+/obj/machinery/recycler/Initialize()
+	. = ..()
+	if(!reagents)
+		create_reagents(RECYCLER_REAGENTS_VOLUME)
+
+/obj/machinery/recycler/Destroy()
+	QDEL_NULL(sound_looping)
+	return ..()
+
+/obj/machinery/recycler/examine(mob/user)
+	. = ..()
+	to_chat(user, "The power light is [(operable() && use_power) ? "on" : "off"].")
+	to_chat(user, "The safety-mode light is [safety_mode ? "on" : "off"].")
+	to_chat(user, "The safety-sensors status light is [emagged ? "off" : "on"].")
+
+/obj/machinery/recycler/power_change()
+	. = ..()
+	if(.)
+		update_sound()
+
+/obj/machinery/recycler/emag_act(remaining_charges, mob/user, emag_source)
+	. = ..()
+	if(!emagged)
+		emagged = TRUE
+		if(safety_mode)
+			safety_mode = FALSE
+			update_icon()
+		playsound(loc, "sparks", 75, 1, -1)
+		to_chat(user, SPAN_NOTICE("You use \the[emag_source] on \the [name]."))
+
+/obj/machinery/recycler/Process()
+	if(operable())
+		output_materials()
+
+/obj/machinery/recycler/dismantle()
+	. = ..()
+	output_materials(TRUE)
+	reagents.splash(loc, reagents.total_volume)
+
+/obj/machinery/recycler/physically_destroyed(skip_qdel)
+	. = ..()
+	output_materials(TRUE)
+	reagents.splash(loc, reagents.total_volume)
+
+/obj/machinery/recycler/on_update_icon()
+	var/active = !(use_power & POWER_USE_OFF) && operable() && !safety_mode
+	icon_state = "[RECYCLER_ICON_STATE_BASE_NAME][active][bloodied? "bld" : ""]"
+
+// This is purely for admin possession !FUN!.
+/obj/machinery/recycler/Bump(var/atom/movable/AM)
+	..()
+	if(AM)
+		Bumped(AM)
+
+//Handle new throwing system
+/obj/machinery/recycler/hitby(atom/movable/AM, datum/thrownthing/TT)
+	if(QDELETED(AM))
+		return
+	if(!can_process_input(AM))
+		return ..()
+	process_input(AM)
+
+/obj/machinery/recycler/Bumped(atom/movable/AM)
```

Comment:
Maybe add identical behavior for hitby(), so that things can be recycled by throwing without a full conveyor setup.